code/obj/item/kitchen.dm

```diff
@@ -981,20 +980,29 @@ TRAYS
 /obj/item/platestack
 	name = "Stack of Plates"
 	desc = "It's a stack of plates"
-	icon = 'icons/obj/foodNdrink/platestack.dmi' //temporary in case of the dmi being different on the live version :)
+	icon = 'icons/obj/foodNdrink/platestack.dmi' // temporary in case of the dmi being different on the live version :)
 	inhand_image_icon = 'icons/obj/foodNdrink/platestackinhand.dmi'
 	icon_state = "platestack1"
 	item_state = "platestack1"
-	var/platenum = 1 //used for targeting icon_states
+	w_class = 4 // why the fuck would you put a stack of plates in your backpack, also prevents shenanigans
+	var/platenum = 1 // used for targeting icon_states
+	var/platemax = 8
+	#if ASS_JAM
+		var/platemax = 13
+	#endif
+
+	proc/update_icon(mob/user as mob)
+		src.icon_state = "platestack[src.platenum]"
+		src.item_state = "platestack[src.platenum]"
+		user.update_inhands()
 
 	attackby(obj/item/weapon as obj,mob/user as mob)
 		if(istype(weapon,/obj/item/plate) && !(istype(weapon,/obj/item/plate/tray)))
 			var/obj/item/plate/p = weapon
 			if(!p.ordered_contents.len)
-				if(!(platenum >= 7))
+				if(!(platenum >= platemax))
```

Comment:
this looks to delete the plates every single attackby call - i think you meant 1 there?

Reply:
The change here is only replacing the original hardcoded limit of 7 with a variable - it evaluates the same as it did already and it still works